src/ns-support.c

```diff
@@ -345,6 +349,20 @@ void sc_create_or_join_ns_group(struct sc_ns_group *group)
 		if (prctl(PR_SET_PDEATHSIG, SIGINT, 0, 0, 0) < 0) {
 			die("cannot set parent process death notification signal to SIGINT");
 		}
+		// Check that parent process is still alive. If this is the case then
+		// we can reliably rely on the PR_SET_PDEATHSIG signal to wake us up
+		// from eventfd_read() below.
+		debug("ensuring that parent process is still alive");
+		if (kill(parent, 0) < 0) {
+			switch (errno) {
+			case ESRCH:
+				debug("parent process has already terminated");
+				abort();
+			default:
+				die("cannot ensure that parent process is still alive");
+				break;
+			}
+		}
```

Comment:
This is definitely better than no check, but this check is not perfect. The parent could die and the kernel could assign the parent pid to a new process. Perhaps it would be better to simply timeout. The parent is supposed to send that signal quite fast... if the child doesn't get that in say, 10 seconds, just debug("parent process did not signal with 10 seconds"); abort();